CodeQL Analysis fails due to a compilation error on example

CodeQL Analysis fails due to  a compilation error on example folder.
The error is the folloring:

```
[ 96%] Building CXX object examples/NetworkDynamicsExample/CMakeFiles/network_dynamics_example.dir/network_dynamics_example.cpp.o
/home/runner/work/CXXGraph/CXXGraph/examples/NetworkDynamicsExample/network_dynamics_example.cpp: In function ‘int main()’:
/home/runner/work/CXXGraph/CXXGraph/examples/NetworkDynamicsExample/network_dynamics_example.cpp:29:48: error: invalid initialization of reference of type ‘const std::vector<int>&’ from expression of type ‘const unsigned int’
   29 |     const std::vector<int>& degrees = nodePair.second;
      |                                       ~~~~~~~~~^~~~~~
gmake[2]: *** [examples/NetworkDynamicsExample/CMakeFiles/network_dynamics_example.dir/build.make:79: examples/NetworkDynamicsExample/CMakeFiles/network_dynamics_example.dir/network_dynamics_example.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2477: examples/NetworkDynamicsExample/CMakeFiles/network_dynamics_example.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```

Please correct it.

--- a/.gitignore
+++ b/.gitignore
@@ -37,3 +37,4 @@ build/
 packaging/
 # ignore vscode files
 .vscode
+.code-workspace

--- a/.gitignore
+++ b/.gitignore
@@ -38,3 +38,4 @@ packaging/
 # ignore vscode files
 .vscode
 .code-workspace
+CXXGraph.code-workspace

--- a/examples/NetworkDynamicsExample/network_dynamics_example.cpp
+++ b/examples/NetworkDynamicsExample/network_dynamics_example.cpp
@@ -26,10 +26,9 @@ int main() {
   auto degreeMatrix = graph.getDegreeMatrix();
   for (const auto& nodePair : *degreeMatrix) {
     const CXXGraph::shared<const CXXGraph::Node<int>>& node = nodePair.first;
-    const std::vector<int>& degrees = nodePair.second;
+    const unsigned int degree = nodePair.second;
 
-    std::cout << "Node: " << node->getId() << ", Degree: " << degrees[0]
-              << "\n";
+    std::cout << "Node: " << node->getId() << ", Degree: " << degree << "\n";
   }
   auto laplacianMatrix = graph.getLaplacianMatrix();
   for (const auto& nodePair : *laplacianMatrix) {

